frontend/src/components/common/Link.tsx

```diff
@@ -28,14 +30,39 @@ export interface LinkObjectProps extends LinkBaseProps {
   [prop: string]: any;
 }
 
+function KubeObjectLink(props: { kubeObject: KubeObject; [prop: string]: any }) {
+  const { kubeObject, ...otherProps } = props;
+
+  const client = useQueryClient();
+  const { namespace, name } = kubeObject.metadata;
+  const endpoint = useEndpoints(kubeObject._class().apiEndpoint.apiInfo, kubeObject.cluster);
+
+  return (
+    <MuiLink
+      onPointerEnter={() => {
+        client.setQueryData(
+          kubeObjectQueryKey({
+            cluster: kubeObject.cluster,
+            endpoint,
+            namespace,
+            name,
+          }),
+          kubeObject
+        );
```

Comment:
I need to clarify something here. Is this resulting in pulling data from the server on "mouse-enter"? If so, I am not sure this is a great idea since we may hover through several Links till we click one.
Regardless, would it work if we set the data on-click instead? Or is this not compatible with the `to=` attribute here?